main.ts

```diff
@@ -1,135 +1,110 @@
+		};
+	}
+}
 export default class SmarterMDhotkeys extends Plugin {
 	async onload() {
-		this.addCommand({
-			id: "smarter-asterisk-bold",
-			name: "Smarter Bold",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("**", "**", editor),
-		});
-		this.addCommand({
-			id: "smarter-underscore-bold",
-			name: "Smarter Underscore Bold",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("__", "__", editor),
-		});
-		this.addCommand({
-			id: "smarter-asterisk-italics",
-			name: "Smarter Italics",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("*", "*", editor),
-		});
-		this.addCommand({
-			id: "smarter-underscore-italics",
-			name: "Smarter Underscore Italics",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("_", "_", editor),
-		});
-		this.addCommand({
-			id: "smarter-comments",
-			name: "Smarter Comments",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("%%", "%%", editor),
-		});
-		this.addCommand({
-			id: "smarter-inline-code",
-			name: "Smarter Inline Code",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("`", "`", editor),
-		});
-		this.addCommand({
-			id: "smarter-highlight",
-			name: "Smarter Highlight",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("==", "==", editor),
-		});
-		this.addCommand({
-			id: "smarter-strikethrough",
-			name: "Smarter Strikethrough",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("~~", "~~", editor),
-		});
-		this.addCommand({
-			id: "smarter-wikilink",
-			name: "Smarter wikilink (internal link)",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("[[", "]]", editor),
-		});
-		this.addCommand({
-			id: "smarter-md-link",
-			name: "Smarter Markdown Link",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("[", "]()", editor),
+		COMMANDS.forEach((command) => {
+			const { id, name, before, after } = command;
+			this.addCommand({
+				id,
+				name,
+				editorCallback: (editor) =>
+					this.expandAndWrap(before, after, editor),
+			});
 		});
 
-		console.log ("Smarter MD Hotkeys loaded.");
+		console.log("Smarter MD Hotkeys loaded.");
 	}
 
 	async onunload() {
-		console.log ("Smarter MD Hotkeys unloaded.");
+		console.log("Smarter MD Hotkeys unloaded.");
 	}
 
 	expandAndWrap(beforeStr: string, afterStr: string, editor: Editor): void {
+		// Arrow functions are simpler
+		// No need to specify the return-type, Typescript can infer it
+		const offToPos = (offset: number) => editor.offsetToPos(offset);
 
-		function offToPos(offset: number): EditorPosition {
-			return editor.offsetToPos(offset);
-		}
-		function WordUnderCursor (ep: EditorPosition): Range{
+		// Return type is infered

```

Comment:
what does it mean? also wouldn't adding the return type basically be s form of commenting?

Reply:
Typescript knows what type is being returned because both of the possible things it can return have the type `EditorRange`.

Yes, adding the return type is certianly useful, especially if TS cannot infer it accurately. But if you hover over the function name, it will tell you already
![image](https://user-images.githubusercontent.com/70717676/142775773-7ebee102-a890-4a4e-b571-902b3bbc1941.png)